src/datepicker/datepicker-input.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  Directive,
+  Input,
+  ComponentRef,
+  ElementRef,
+  ViewContainerRef,
+  Renderer,
+  ComponentFactoryResolver,
+  NgZone,
+  TemplateRef,
+  forwardRef
+} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+
+import {NgbDate} from './ngb-date';
+import {NgbDatepicker} from './datepicker';
+import {DayTemplateContext} from './datepicker-day-template-context';
+import {NgbDateParserFormatter} from './ngb-date-parser-formatter';
+
+import {positionElements} from '../util/positioning';
+
+const NGB_DATEPICKER_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => NgbInputDatepicker),
+  multi: true
+};
+
+@Directive({
+  selector: 'input[ngbDatepicker]',
+  exportAs: 'ngbDatepicker',
+  host: {'(change)': 'manualDateChange($event.target.value)', '(keyup.esc)': 'close()'},
+  providers: [NGB_DATEPICKER_VALUE_ACCESSOR]
+})
+export class NgbInputDatepicker implements ControlValueAccessor {
+  private _cRef: ComponentRef<NgbDatepicker> = null;
+  private _model: NgbDate;
+  private _zoneSubscription: any;
+
+  /**
+   * Reference for the custom template for the day display
+   */
+  @Input() dayTemplate: TemplateRef<DayTemplateContext>;
+
+  /**
+   * First day of the week, 0=Sun, 1=Mon, etc.
+   */
+  @Input() firstDayOfWeek: number;
+
+  /**
+   * Callback to mark a given date as disabled
+   */
+  @Input() markDisabled: (date: {year: number, month: number, day: number}) => boolean;
```

Comment:
Couldn't we create an interface for this structure. What would be the disadvantage of doing so? It would make our code easier to read (IMO), and the client code safer too, because it could use that interface too, and thus avoid typos like `seconds` instead of `second`?